### PR TITLE
added bulk support for multi-band files

### PR DESCRIPTION
**Pull request by RousseauLambertLP**
added support for bulk insert in ES when there are more then one band per file.

### CODE REVIEW

File: geomet_data_registry/tileindex/elasticsearch_.py [hunk 1]
```diff
@@ -240,6 +240,58 @@ def add(self, identifier, data):
...
+                    'index': {
+                        '_index': self.name,
+                        '_type': '_doc'
+                    }
+                }
+                op_dict['index']['_id'] = doc['properties']['identifier']
+                bulk_data.append(op_dict)
+                bulk_data.append(doc)
+            r = self.es.bulk(index=self.name, body=bulk_data,
+                             pipeline='gdr_register_datetime')
```
Comment (tomkralidis) [inline geomet_data_registry/tileindex/elasticsearch_.py]: call this `gdr_register_datetime_pipeline`

File: geomet_data_registry/tileindex/elasticsearch_.py [hunk 2]
```diff
@@ -240,6 +240,58 @@ def add(self, identifier, data):
...
+        :param data: GeoJSON dict
+
+        :returns: list of dict {layer_id: HTTP status code}
+        """
+
+        status_code = None
+
+        LOGGER.debug('Starting bulk add')
+        try:
+            # FIXME: refresh='wait_for' essentially blocks further processing
```
Comment (tomkralidis) [inline geomet_data_registry/tileindex/elasticsearch_.py]: Do we need this `FIXME` here?

File: geomet_data_registry/tileindex/elasticsearch_.py [hunk 3]
```diff
@@ -240,6 +240,58 @@ def add(self, identifier, data):
...
+                tmp_dict = {}
+                tmp_dict[lyr_id] = lyr_status
+                status_arr.update(tmp_dict)
+
+        except Exception as err:
+            LOGGER.exception('Error bulk indexing: {}'.format(err))
+            return 500
+
+
+        return status_arr
```
Comment (tomkralidis) [inline geomet_data_registry/tileindex/elasticsearch_.py]: Should this not be a status code like the `add` function?

File: geomet_data_registry/tileindex/elasticsearch_.py (hunk 2 above)
Comment (Dukestep) [reply geomet_data_registry/tileindex/elasticsearch_.py]: We can delete the FIXME. Sorry, should've done this in the pipeline MR.

File: geomet_data_registry/tileindex/elasticsearch_.py (hunk 1 above)
Comment (Dukestep) [reply geomet_data_registry/tileindex/elasticsearch_.py]: If we change it here we will have to change the pipeline name in `ElasticsearchTileIndex.setup()`, `ElasticsearchTileIndex.teardown()` and `ElasticsearchTileIndex.add()`.

File: geomet_data_registry/tileindex/elasticsearch_.py (hunk 3 above)
Comment (Dukestep) [reply geomet_data_registry/tileindex/elasticsearch_.py]: We need the result status for each document contained in the bulk_add so we can update the store accordingly (if it's an update don't add to the count, if it's a newly created document add to store count). If we just return a status code there's no way if knowing what variables were added or simply updated!

Comment (RousseauLambertLP) [reply geomet_data_registry/tileindex/elasticsearch_.py]: agree with @Dukestep we need to have a status per layer in geomet for the count.

File: geomet_data_registry/tileindex/elasticsearch_.py (hunk 2 above)
Comment (RousseauLambertLP) [reply geomet_data_registry/tileindex/elasticsearch_.py]: fix in merge request, marking this as resolved.

File: geomet_data_registry/tileindex/elasticsearch_.py (hunk 1 above)
Comment (RousseauLambertLP) [reply geomet_data_registry/tileindex/elasticsearch_.py]: I can do the change and update the names in those files as well. 

@tomkralidis should I go on and update those other files ?

**Pull request merged**